storage/replica.go

```diff
@@ -2111,6 +2111,13 @@ func (r *Replica) loadSystemConfigSpan() ([]roachpb.KeyValue, []byte, error) {
 	return kvs, config.SystemConfig{Values: kvs}.Hash(), nil
 }
 
+// needsSplitBySize returns true if the size of the range requires it
+// to be split.
+func (r *Replica) needsSplitBySize() bool {
+	maxBytes := r.GetMaxBytes()
+	return maxBytes > 0 && r.stats.KeyBytes+r.stats.ValBytes > maxBytes
```

Comment:
Hmm, how was it working in `maybeAddToSplitQueue` then? (and I see I only did half the job of extracting this code from that method) I guess it was protected by virtue of being in the processRaft goroutine? Synchronized this correctly and un-embedded `rangeStats.MVCCStats` to discourage direct unsynchronized access to members.